src/components/Layout/LayoutHeader.tsx

```diff
@@ -0,0 +1,156 @@
+import { useState, useRef } from 'react';
+import { useRouter } from 'next/router';
+import { Button, Flex, View, VisuallyHidden } from '@aws-amplify/ui-react';
+import classNames from 'classnames';
+import { DEFAULT_PLATFORM, PLATFORMS, Platform } from '@/data/platforms';
+import {
+  ALGOLIA_API_KEY,
+  ALGOLIA_INDEX_NAME,
+  ALGOLIA_APP_ID
+} from '../../constants/algolia';
+import { IconMenu, IconDoubleChevron } from '@/components/Icons';
+import { Menu } from '@/components/Menu';
+import type { HeadingInterface } from '@/components/TableOfContents/TableOfContents';
+import { PlatformNavigator } from '@/components/PlatformNavigator';
+import flatDirectory from 'src/directory/flatDirectory.json';
+import { DocSearch } from '@docsearch/react';
+import '@docsearch/css';
+import { PageLastUpdated } from '../PageLastUpdated';
+import Feedback from '../Feedback';
+import RepoActions from '../Menu/RepoActions';
+import { usePathWithoutHash } from '@/utils/usePathWithoutHash';
+
+export const LayoutHeader = ({
+  hasTOC = true,
+  tocHeadings,
+  pageType = 'inner',
+  platform,
+  showLastUpdatedDate = true
+}: {
+  hasTOC?: boolean;
+  tocHeadings: HeadingInterface[];
+  pageType?: 'home' | 'inner';
+  platform?: Platform;
+  showLastUpdatedDate: boolean;
+}) => {
+  const [menuOpen, toggleMenuOpen] = useState(false);
```

Comment:
Can you take a look at this functionality? The opening/closing of the menu works on mobile, but closing the menu when you navigate to a page doesn't occur now (currently on prod if you are in the mobile view, you click on a page in the menu, the menu closes). `toggleMenuOpen` is part of the context passed down from LayoutProvider, maybe we need to import that context here like we do in [MenuItem](https://github.com/aws-amplify/docs/blob/main/src/components/Menu/MenuItem.tsx#L43)